core/src/main/scala/org/apache/spark/rdd/CheckpointRDD.scala

```diff
@@ -59,9 +61,6 @@ class CheckpointRDD[T: ClassTag](sc: SparkContext, val checkpointPath: String)
     Array.tabulate(numPartitions)(i => new CheckpointRDDPartition(i))
   }
 
-  checkpointData = Some(new RDDCheckpointData[T](this))
-  checkpointData.get.cpFile = Some(checkpointPath)
```

Comment:
I removed this because the checkpoint data here isn't really used; it's only there for `getCheckpointFile` to return the right path, which we can just provide by overriding it ourselves.